modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/TransactionalPublishingTestCase.java

```diff
@@ -534,4 +538,55 @@ public void topicPublishingWithoutSubsribers() throws XPathExpressionException,
             publisherClient1.stopClient();
         }
     }
+
+    /**
+     * Test flow control when publishing more than 1000 messages with transactional session enabled
+     * Related JIRA: https://wso2.org/jira/browse/MB-1928
+     *
+     * @throws XPathExpressionException
+     * @throws AndesClientConfigurationException
+     * @throws IOException
+     * @throws AndesClientException
+     * @throws JMSException
+     * @throws NamingException
+     * @throws AutomationUtilException
+     * @throws AndesAdminServiceBrokerManagerAdminException
+     */
+    @Test(groups = {"wso2.mb", "queue", "transactions"}, description = "Test queue publisher without subscribers")
+    public void queuePublishingWithoutSubscribers()
+            throws XPathExpressionException, AndesClientConfigurationException, IOException, AndesClientException,
+                   JMSException, NamingException, AutomationUtilException,
+                   AndesAdminServiceBrokerManagerAdminException {
+
+        int messageCount = 2000;
+        String queueName = "pubWithoutSubQueue";
+
+        //Logging in
+        LoginLogoutClient loginLogoutClientForAdmin = new LoginLogoutClient(super.automationContext);
+        String sessionCookie = loginLogoutClientForAdmin.login();
+
+        //Creating admin client
+        AndesAdminClient admin = new AndesAdminClient(super.backendURL, sessionCookie);
+
+        //Creating a queue to publish messages
+        admin.createQueue(queueName);
+
+        //Creating publisher configuration
+        AndesJMSPublisherClientConfiguration publisherConfig =
+                new AndesJMSPublisherClientConfiguration(getAMQPPort(), ExchangeType.QUEUE, queueName);
+        publisherConfig.setTransactionalSession(true);
+        publisherConfig.setNumberOfMessagesToSend(messageCount);
+
+        //Creating publisher
+        AndesClient publisherClient = new AndesClient(publisherConfig, true);
+        publisherClient.startClient();
+        AndesClientUtils.sleepForInterval(5000);
```

Comment:
So we're actually browsing messages after 5 seconds of publishing. This will not be enough. We need to wait for more than 2 minutes which is the timeout for publisher flow controling.